packages/core/src/application.ts

```diff
@@ -61,9 +63,74 @@ export class Application extends Context {
    */
   controller(controllerCtor: ControllerClass, name?: string): Binding {
     name = name || controllerCtor.name;
-    return this.bind(`controllers.${name}`)
+    return this.bind(`${CoreBindings.CONTROLLERS}.${name}`)
       .toClass(controllerCtor)
-      .tag('controller');
+      .tag(CoreBindings.CONTROLLERS_TAG);
+  }
+
+  /**
+   * Register a booter class / array of classes with this application.
+   *
+   * @param booterCls {Function | Function[]} The booter class (constructor function).
+   * @param {string=} name Optional booter name, defaults to the class name.
+   * Ignored is cls is an Array and the name defaults to the class name.
+   * @return {Binding | Binding[]} The newly created binding(s), you can use the
+   * reference to further modify the binding, e.g. lock the value to prevent
+   * further modifications.
+   *
+   * ```ts
+   * class MyBooter implements Booter {}
+   * app.booter(MyBooter);
+   * ```
+   */
+  booter(booterCls: Constructor<Booter>, name?: string): Binding;
+  booter(booterCls: Constructor<Booter>[]): Binding[];
+  booter(
+    booterCls: Constructor<Booter> | Constructor<Booter>[],
+    name?: string,
+    // tslint:disable-next-line:no-any
+  ): any {
+    if (Array.isArray(booterCls)) {
+      return booterCls.map(cls => this._bindBooter(cls));
+    } else {
+      return this._bindBooter(booterCls, name);
+    }
+  }
+
+  /**
+   *
+   * @param booterCls Constructor<Booter> A Booter Class
+   * @param [name] Name the Booter Class should be bound to
+   */
+  private _bindBooter<T extends Booter>(
+    booterCls: Constructor<T>,
+    name?: string,
+  ): Binding {
+    name = name || booterCls.name;
```

Comment:
I suggest we also add a tag such as `name:<my-name>` as a tag.

Reply:
I didn't add the name as a tag because I wasn't sure if it's possible for us to filter based on multiple tags combined together. (BOOTER_TAG && (name:Booter1 || name:Booter2)). 

I did however switch to filtering by the name a user uses to bind the Booter using `app.booter(Booter, name)` and the default name happens to be the class name.